projects/plugins/jetpack/modules/woocommerce-analytics/classes/class-jetpack-woocommerce-analytics-my-account.php

```diff
@@ -0,0 +1,301 @@
+<?php
+/**
+ * Jetpack_WooCommerce_Analytics_My_Account
+ *
+ * @package automattic/jetpack
+ * @author  Automattic
+ */
+
+/**
+ * Bail if accessed directly
+ */
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Jetpack_WooCommerce_Analytics_My_Account
+ * Filters and Actions added to My Account pages to perform analytics
+ */
+class Jetpack_WooCommerce_Analytics_My_Account {
+
+	use Jetpack_WooCommerce_Analytics_Trait;
+
+	/**
+	 * Jetpack_WooCommerce_Analytics_My_Account constructor.
+	 */
+	public function __construct() {
+
+		add_action( 'woocommerce_account_content', array( $this, 'track_tabs' ) );
+		add_action( 'woocommerce_account_content', array( $this, 'track_logouts' ) );
+		add_action( 'woocommerce_customer_save_address', array( $this, 'track_save_address' ), 10, 2 );
+		add_action( 'wp_head', array( $this, 'trigger_queued_events' ) );
+		add_action( 'wp', array( $this, 'track_add_payment_method' ) );
+		add_action( 'wp', array( $this, 'track_delete_payment_method' ) );
+		add_action( 'woocommerce_save_account_details', array( $this, 'track_save_account_details' ) );
+		add_filter( 'woocommerce_my_account_my_orders_actions', array( $this, 'add_initiator_prop_to_my_account_action_links' ) );
+		add_action( 'woocommerce_cancelled_order', array( $this, 'track_order_cancel_event' ), 10, 0 );
+		add_action( 'before_woocommerce_pay', array( $this, 'track_order_pay_event' ) );
+		add_action( 'woocommerce_before_account_orders', array( $this, 'add_initiator_prop_to_order_urls' ), 9 );
+		add_filter( 'query_vars', array( $this, 'add_initiator_param_to_query_vars' ) );
+	}
+
+	/**
+	 * Track my account tabs, we only trigger an event if a tab is viewed.
+	 *
+	 * We also track other events here, like order number clicks, order action clicks,
+	 * address clicks, payment method add and delete.
+	 */
+	public function track_tabs() {
+		global $wp;
+
+		// WooCommerce keeps a map of my-account endpoints keys and their custom permalinks.
+		$core_endpoints = WC()->query->get_query_vars();
+
+		if ( ! empty( $wp->query_vars ) ) {
+
+			foreach ( $wp->query_vars as $key => $value ) {
+				// we skip pagename
```

Comment:
Nitpick about this comment, we should use a capital letter at the start and full stop at the end
```suggestion
				// We skip pagename.
```